### DISCUSSION TITLE
Welcome to v3.0.0 release

### DISCUSSION THREAD

**Discussion by A7Lavinraj** [A7Lavinraj/assistant.nvim#46] [Announcements]
# 🎉 Release v3.0.0 Discussion

Hi everyone! 🚀  

We're thrilled to announce the release of **v3.0.0** of this plugin! This version brings significant updates, improvements, and bug fixes based on community feedback and our goals to enhance the overall user experience.

## 🔥 What's New in v3.0.0?
- **New interactive UI:** Introducing brand new plugin UI for better UX and productivity.
- **New parallel task scheduler:** Now you can run as many tasks as you want without looking at CPU over loading thanks to new implemented task scheduler.
- **New actions bar:** Now you can get a live feedback for your actions(track only few actions).
- **More flexibility over configuration:** Now you can configure the plugin to the next level.

## 👀 Share Your Feedback  
We'd love to hear your thoughts on this release! Let us know:
- How is your experience with the new features?  
- Are there any areas where you feel we can improve?  
- Any bugs or unexpected issues you've encountered?  

Your feedback helps us shape the future of this project, so don’t hold back!

## 📖 Resources
- [Documentation](https://github.com/A7Lavinraj/assistant.nvim/blob/main/README.md)  

![demo](https://github.com/user-attachments/assets/3b8144b3-5597-48ad-a41c-f1a684eb7a69)

**Comment by suzhiking** [comment]
How does this plugin compare to https://github.com/xeluxee/competitest.nvim?

**Reply by A7Lavinraj** [reply]
- It has scheduling system for running tests
- centralized saved data
- easy to use with nicer ui
- it shows proper compilation error
- also has continous sample fetching which means u do not have to run command to start sample fetching 

And one important thing it will also have stress and validation testing in future

**Reply by suzhiking** [reply]
Thanks. How can I receive test cases from Competitive Companion like competitest.nvim?

**Reply by A7Lavinraj** [reply]
Just open neovim and click on browser extension and that's it

**Comment by shaitanu** [comment]
I am facing very weird error. I am able to parse problems from firefox if i first parse at least one problem from my Brave Browser.

**Reply by A7Lavinraj** [reply]
OK stay tuned. I will add video tutorial as soon as possible 😄

**Reply by shaitanu** [reply]
thanks! and btw your dotfiles config for nvim is broken sh install.sh was also faiiling.

**Reply by A7Lavinraj** [reply]
Its not updated yet, but would be update soon

**Reply by shaitanu** [reply]
yeah will wait for that only now. I liked conqueror of completion but cant use it with lazyvim. I am having very hard time setting up lsp,dap and formatter for codeforces cp.

**Reply by A7Lavinraj** [reply]
Don't worry I will make video about whole setup for cp with and without plugins. Stay tuned 😁

**Reply by A7Lavinraj** [reply]
## CHECKOUT THIS YOUTUBE TUTORIAL!

[![YouTube](http://i.ytimg.com/vi/RuLOxD8_KXo/hqdefault.jpg)](https://www.youtube.com/watch?v=RuLOxD8_KXo)

**Comment by rmenai** [comment]
This plugin looks great! It would be awesome if it has template support for each language. Also an option to set up a default language would be great!

**Reply by A7Lavinraj** [reply]
Nice one! @rmenai. I am listing this feature to v4.0.0 which will be the biggest release for this plugin so far. Thanks for supporting 😁

**Comment by Enoch-Tadesse** [comment]
How do I create my own test cases? I used 'c' to create a new test case but when I press either 'i' or 'e', it throws error

**Reply by A7Lavinraj** [reply]
Can you please upload a video. It will be easy to debug your issue

**Comment by Enoch-Tadesse** [comment]
Do you want me to upload the video here on Gmail or on github?

**Reply by A7Lavinraj** [reply]
Github of course 😅

**Reply by Enoch-Tadesse** [reply]
[Uploading Screencast from 2025-02-02 15-38-02.webm…]()

**Reply by Enoch-Tadesse** [reply]
> Github of course 😅

![Screenshot from 2025-02-02 15-42-08](https://github.com/user-attachments/assets/51d00714-99d8-4846-88b1-f676f50d38cf)

**Reply by Enoch-Tadesse** [reply]
https://github.com/Enoch-Tadesse/nvim

lua/custom/plugins.lua

**Reply by Enoch-Tadesse** [reply]
The error occurred because I was using nvchad 2.0, I upgraded to nvchad 2.5. Thanks for your effort.

**Reply by A7Lavinraj** [reply]
I glad your error get resolved
